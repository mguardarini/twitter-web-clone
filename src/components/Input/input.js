import React from 'react';
import TextField from '@material-ui/core/TextField';
import clsx from 'clsx';
import useStyles from './style';

export default function InputComponent(props) {
    const classes = useStyles();
    const height=28;
    const width=196;
    return (
        <TextField
            id="outlined-dense"
            className={clsx(classes.textfield)}
            label={props.placeholder}
            variant="outlined"
            inputProps={{
                style: {
                  height,
                  width,
                  padding: '10px',
                  fontSize: "11px"

                },
                
            }}
            InputLabelProps={{
                style: {
                  textOverflow: 'ellipsis',
                  whiteSpace: 'nowrap',
                  overflow: 'hidden',
                  width: '100%',
                  fontSize:'14px'
                } }}
        ></TextField>
    )
}
  