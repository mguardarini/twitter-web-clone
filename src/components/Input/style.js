import { makeStyles} from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
    textfield: {
      margin: theme.spacing(1),
      marginBottom:'0px',
      fontStyle: 'normal',
      fontFamily:'Segoe UI',
    }
}));
  
export default useStyles;