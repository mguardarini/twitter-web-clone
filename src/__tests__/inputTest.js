// import dependencies
import React from 'react'
import {render, fireEvent} from '@testing-library/react'
import Input from '../../src/components/Input/input'
 
describe('<Input /> spec', () => {
    it('renders the component', () => {
      const container = render(<Input placeholder='Nome' />)
      expect(container.firstChild).toMatchSnapshot()
      expect(container.findByLabelText("Nome"))
     })
     
     it('find component by label and write there', () => {
      const {getByLabelText} = render(<Input placeholder='Nome' />);    
      const input = getByLabelText('Nome');
      input.value = 'Test component'
    })
});