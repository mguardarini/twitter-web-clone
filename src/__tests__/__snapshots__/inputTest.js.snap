// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> spec find component by label 1`] = `undefined`;

exports[`<Input /> spec renders the component 1`] = `undefined`;
